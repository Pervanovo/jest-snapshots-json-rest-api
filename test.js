/* eslint-env jest */

const serializer = require('./')

expect.addSnapshotSerializer(serializer)

test('should not serialize non-HTTP response', () => {
  const object = { foo: 'bar' }
  expect(object).toMatchSnapshot()
})

test('should serialize HTTP response', () => {
  const object = { status: 200, body: { hello: 'world' } }
  expect(object).toMatchSnapshot()
})

test('should serialize HTTP Buffer response', () => {
  const object = { statusCode: 200, body: Buffer.from('{ "Hello": "World" }') }
  expect(object).toMatchSnapshot()
})

test('should serialize LightMyRequest HTTP response', () => {
  const object = { statusCode: 200, body: '{ "Hello": "World" }' }
  expect(object).toMatchSnapshot()
})
