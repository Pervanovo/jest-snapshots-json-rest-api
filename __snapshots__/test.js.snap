// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not serialize non-HTTP response 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`should serialize HTTP Buffer response 1`] = `
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "properties": {
    "Hello": {
      "type": "string"
    }
  },
  "type": "object"
}
`;

exports[`should serialize HTTP response 1`] = `
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "properties": {
    "hello": {
      "type": "string"
    }
  },
  "type": "object"
}
`;

exports[`should serialize LightMyRequest HTTP response 1`] = `
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "properties": {
    "Hello": {
      "type": "string"
    }
  },
  "type": "object"
}
`;
